src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/JobSchedulerPlugin.java

```diff
@@ -75,8 +77,10 @@ public JobSchedulerPlugin() {
                            ResourceWatcherService resourceWatcherService, ScriptService scriptService,
                            NamedXContentRegistry xContentRegistry, Environment environment,
                            NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry) {
-        this.scheduler = new JobScheduler(threadPool);
-        this.sweeper = initSweeper(environment.settings(), client, clusterService, threadPool, xContentRegistry, this.scheduler);
+        this.jobSchedulerLock = new LockService(client.admin().indices(), client);
+        this.scheduler = new JobScheduler(threadPool, jobSchedulerLock);
```

Comment:
Using a mix of jobSchedulerLock and this.jobSchedulerLock (in JobScheduler and initSweeper call)